src/app/core/services/api/service.ts

```diff
@@ -344,4 +344,19 @@ export class ApiService {
     const url = `${this._newRestRoot}/constrainttemplates`;
     return this._http.get<ConstraintTemplate[]>(url);
   }
+
+  createConstraintTemplate(template: ConstraintTemplate): Observable<ConstraintTemplate> {
```

Comment:
Can you extract it to a dedicated service? We've started in the past process of refactoring this big API service but it was never finished.